rasa/core/training/interactive.py

```diff
@@ -1569,6 +1600,7 @@ def visualisation_png(request):
 def run_interactive_learning(
     file_importer: TrainingDataImporter,
     skip_visualization: bool = False,
+    conversation_id: Text = uuid.uuid4().hex,
```

Comment:
Why do we need to set this here as well as [here](https://github.com/RasaHQ/rasa/pull/5243/files#diff-df892daf3511544c9c31f911b4ffd0a1R57)?

Reply:
Would be enough to only set it once. The same holds for `skip_visualization` though and that was set in both places, so we went with the same redundancy for `conversation_id` to keep it consistent.

Also, I think it makes sense because of this:
- We need the default value in the argument parser because then `rasa interactive --help` will display it.
- If the code ever changes such that `run_interactive_learning` will also be called elsewhere than via `rasa interactive` it is useful to have the default value set also here in the function.

Reply:
> We need the default value in the argument parser because then rasa interactive --help will display it.

Hm, this is a good point. But also makes me think that the default might be e.g. 9a1dcae1f3ad41b791eb7a6d9e7a57c7 and i'd expect to be able to query the tracker for that, when in reality by the time i did the actual command, it would have changed. 

> If the code ever changes such that run_interactive_learning will also be called elsewhere than via rasa interactive it is useful to have the default value set also here in the function.

Yes, definitely agree with this.

Reply:
One thing to add, I don't know if we need to change this functionality here or not, but wanted to point out that the default `conversation-id` for the [other PR](https://github.com/RasaHQ/rasa/pull/5117) is going to be `"default"`, so, not sure if the difference in functionality of the same parameter for the two commands is odd.

Reply:
Maybe we should have the default value only in this function then, not in the parser, and for the parser manually add a sentence like "Defaults to a random uuid"?

Reply:
About the second point: This is how it used to be before. Previously, there were no command line arguments and instead the id was always `default` for `rasa shell` and a random uuid for `rasa interactive` (see also here: [issue](https://github.com/RasaHQ/rasa/issues/3975))
I don't know if there is a specific reason why they were chosen like this. Do you think we should change this one to also be `default`?

Reply:
Oh, that's strange. My only thought is that maybe it's possible that each time you started a new conversation in interactive mode, it got a different UUID, making each conversation different? 🤔 @wochinge what do you think? If we keep it as this implementation, i think `Maybe we should have the default value only in this function then, not in the parser, and for the parser manually add a sentence like "Defaults to a random uuid"?` sounds good.

Reply:
> My only idea is that maybe it's possible that each time you started a new conversation in interactive mode, it got a different UUID, making each conversation different?

I think this is a good thing. When I run `rasa interactive` I want to start from scratch and not start at a random state. 

> Maybe we should have the default value only in this function then, not in the parser, and for the parser manually add a sentence like "Defaults to a random uuid"?

I'd keep it in the parser and maybe a sentence to the help message. That should make it clear enough and we don't add unnecessary handling of the `None` value to the code.

Reply:
>  When I run rasa interactive I want to start from scratch and not start at a random state.

Pretty sure it always would, though, or the same issue would happen in `shell`

But I'm fine either way.

Reply:
The same thing happens in shell if you have a persistent tracker store

Reply:
Hm, had no idea. Ok @chkoss let's just go with this then:
> I'd keep it in the parser and maybe a sentence to the help message. That should make it clear enough and we don't add unnecessary handling of the None value to the code.

Reply:
Done :) Should be ready to be merged now, right?

If the same thing happens in `shell` with a persistent tracker store, then it would probably be better to make it a random ID there, too, instead of `'default'`. Shall I open an issue for that?

Reply:
@chkoss maybe you can comment on the open PR to that effect? https://github.com/RasaHQ/rasa/pull/5117